tests/components/onvif/test_config_flow.py

```diff
@@ -0,0 +1,508 @@
+            user_input={config_flow.CONF_HOST: config_flow.CONF_MANUAL_INPUT},
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "manual_input"
+
+
+async def test_flow_discovery_ignore_existing_and_abort(hass):
+    """Test that config flow discovery ignores setup devices."""
+    await setup_onvif_integration(hass)
+    await setup_onvif_integration(
+        hass,
+        config={
+            config_flow.CONF_NAME: DISCOVERY[1]["EPR"],
+            config_flow.CONF_HOST: DISCOVERY[1][config_flow.CONF_HOST],
+            config_flow.CONF_PORT: DISCOVERY[1][config_flow.CONF_PORT],
+            config_flow.CONF_USERNAME: "",
+            config_flow.CONF_PASSWORD: "",
+        },
+        unique_id=DISCOVERY[1]["MAC"],
+        entry_id="2",
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user"}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    with patch(
+        "homeassistant.components.onvif.config_flow.get_device"
+    ) as mock_device, patch(
+        "homeassistant.components.onvif.config_flow.wsdiscovery"
+    ) as mock_discovery, patch(
+        "homeassistant.components.onvif.camera.ONVIFHassCamera"
+    ) as mock_camera:
+        setup_mock_onvif_device(mock_device)
+        setup_mock_discovery(mock_discovery, with_name=True, with_mac=True)
+        setup_mock_camera(mock_camera)
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={}
+        )
+
+        # It should skip to manual entry if the only devices are already configured
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "manual_input"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                config_flow.CONF_NAME: NAME,
+                config_flow.CONF_HOST: HOST,
+                config_flow.CONF_PORT: PORT,
+            },
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "auth"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                config_flow.CONF_USERNAME: USERNAME,
+                config_flow.CONF_PASSWORD: PASSWORD,
+            },
+        )
+
+        # It should abort if already configured and entered manually
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+
+
+async def test_flow_manual_entry(hass):
+    """Test that config flow works for discovered devices."""
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user"}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5453535)